src/datasets/formatting/formatting.py

```diff
@@ -304,6 +303,7 @@ def __repr__(self):
         return repr(self.data)
 
     if config.PY_VERSION >= version.parse("3.9"):
+        # merging with the union ("|") operator is supported in Python 3.9+
```

Comment:
You don't use it as far as I can tell

Reply:
This class needs to mimic the built-in dict inside the map function, so it needs to support all of its methods.